internal/service/service.go

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package service
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type Service struct {
+	Host string
+	Port int
+}
+
+func (c Service) EndpointURL(endpoint string) string {
```

Comment:
Please use better descriptive names. Like `GetEndpointUrl` for this one.

Reply:
good point. done.